source/common/conn_pool/conn_pool_base.cc

```diff
@@ -387,18 +392,15 @@ void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view
   } else if (event == Network::ConnectionEvent::Connected) {
     client.conn_connect_ms_->complete();
     client.conn_connect_ms_.reset();
-
     ASSERT(client.state_ == ActiveClient::State::CONNECTING);
     transitionActiveClientState(client, ActiveClient::State::READY);
 
+    // At this point for the mixed ALPN pool client may be deleted. Do not
```

Comment:
nit, I had to read this a couple of times.

```suggestion
    // At this point, for the mixed ALPN pool, the client may be deleted. Do not
```